### PR TITLE
Fix for CET-312

### PR DESCRIPTION
**Pull request by NaveenSreeramachandra**
Datera OS had changed their default behavior to reject unrestricted volume create calls when provisioned into a tenant with restricted IOPS settings. The driver fix in the PR is to do the following:

(1) Send the QoS performance params along with Volume Creation request API itself.
(2) Disable updating the QoS params later by a separate API call.

### CODE REVIEW

File: deploy/kubernetes/release/1.0/csi-datera-1.0.8.yaml [hunk 1]
```diff
@@ -371,7 +371,7 @@ spec:
             capabilities:
               add: ["SYS_ADMIN"]
             allowPrivilegeEscalation: true
-          image: dateraiodev/dat-csi-plugin:v1.0.8
+          image: dateraiodev/dat-csi-plugin:v1.0.8-patch
```
Comment (mittalms) [inline deploy/kubernetes/release/1.0/csi-datera-1.0.8.yaml]: Does it need to be called patch or a minor version bump should be done here.

File: pkg/client/volume.go [hunk 2]
```diff
@@ -301,6 +310,14 @@ func (r *DateraClient) CreateVolume(name string, volOpts *VolOpts, qos bool) (*V
...
 				PlacementMode: volOpts.PlacementMode,
 				ReplicaCount:  int(volOpts.Replica),
+				PerformancePolicy: &dsdk.PerformancePolicy{
+					WriteIopsMax: int(volOpts.WriteIopsMax),
+					ReadIopsMax: int(volOpts.ReadIopsMax),
+					TotalIopsMax: int(volOpts.TotalIopsMax),
+					WriteBandwidthMax: int(volOpts.WriteBandwidthMax),
+					ReadBandwidthMax: int(volOpts.ReadBandwidthMax),
+					TotalBandwidthMax: int(volOpts.TotalBandwidthMax),
+				},
```
Comment (virgilm) [inline pkg/client/volume.go]: This is ok, it should work, but it leaves dead code around.

I think you should:
- remove the boolean QoS parameter from the CreateVolume method in volume.go
- remove the code that uses that parameter below (volume.go:347, as that is dead code now)
- remove the qos parameter when calling CreateVolume

File: pkg/driver/controller.go [hunk 3]
```diff
@@ -303,7 +303,9 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 	}
 	params.Size = size
 	// Create AppInstance/StorageInstance/Volume
-	vol, err := d.dc.CreateVolume(id, params, true)
+	// Fix for CET-312. QoS params sent along with volume creation call
+	// No need to update the performance_policy again
+	vol, err := d.dc.CreateVolume(id, params, false)
```
Comment (virgilm) [inline pkg/driver/controller.go]: This is the only call to CreateVolume, I don't see the point of the QoS parameter anymore.

File: pkg/client/volume.go (hunk 2 above)
Comment (NaveenSreeramachandra) [reply pkg/client/volume.go]: The tricky part here is that - removing QoS alone is not sufficient. There is another condition about empty template in that if condition. I can remove that condition too and run another regression to check if anything breaks.

Comment (virgilm) [reply pkg/client/volume.go]: It's
`	if qos && volOpts.Template == ""
`

So, if qos is false (which is always the case now), then the code at volume.go:347 is dead code.

File: pkg/driver/controller.go (hunk 3 above)
Comment (NaveenSreeramachandra) [reply pkg/driver/controller.go:308]: Removed the QoS param.

File: deploy/kubernetes/release/1.0/csi-datera-1.0.8.yaml (hunk 1 above)
Comment (NaveenSreeramachandra) [reply deploy/kubernetes/release/1.0/csi-datera-1.0.8.yaml:362]: New version 1.0.9 is tagged and released.

**Comment by NaveenSreeramachandra** [comment]
Merging Volume expansion, QoS performance policy inclusions, updated README and go modules changes to Master.

**Pull request merged**